src/main/java/com/google/sps/user/User.java

```diff
@@ -0,0 +1,81 @@
+package com.google.sps.user;
+
+import org.jetbrains.annotations.Nullable;
+
+/** Stores the data related to one user */
+public class User {
+    private final String id, email;
+    private final boolean publicPortfolio;
+    @Nullable 
+    private final String name, selfIntroduction, imgUrl;
+
+    public static class Builder {
+        //Required
+        private final String id, email;
+        //Optional
+        private String name, selfIntroduction, imgUrl;
```

Comment:
Change to final variable? We wouldn't need to set a field more than once.

Reply:
You are right about not changing these fields more than once, but I can't change them to final in the Builder class, because I need to assign values to them outside the constructor. It would result in "cannot assign a value to final variable <variable name>" error. 
See [Effective Java](https://effectivejava.googleplex.com/third-edition), Item 2, page 13. It uses the same pattern, and also omits the final keyword. 
Remark that the fields of the actual User class are final.